### PR TITLE
UPPSF-131 Added parameterised waiting interval in the call to contentResolver (document-store-api) to avoid getting still not existing content there

### PR DESCRIPTION
**Pull request by gkazakov111**

### CODE REVIEW

File: helm/content-collection-unfolder/app-configs/content-collection-unfolder_delivery.yaml [hunk 1]
```diff
@@ -11,3 +11,4 @@ env:
   RELATIONS_RESOLVER_URI: http://relations-api:8080/contentcollection/{uuid}/relations
   RELATIONS_RESOLVER_HEALTH_URI: http://relations-api:8080/__health
   Q_WRITE_TOPIC: PostPublicationEvents
+  REQUEST_TIMEOUT: 2
```
Comment (rnov) [inline helm/content-collection-unfolder/app-configs/content-collection-unfolder_delivery.yaml]: missing eol

File: resolver/contentResolver.go [hunk 2]
```diff
@@ -70,3 +74,78 @@ func (cr *defaultContentResolver) callContentResolverApp(diffUuids []string, tid
...
+
+	bodyAsBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("Could not read response after calling [%v], transaction_id=[%v], error was: [%v]", cr.contentResolverAppURI, tid, err.Error())
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("Call to [%v] for transaction_id=[%v], responded with error statusCode [%d], error was: [%v]", cr.contentResolverAppURI, tid, resp.StatusCode, string(bodyAsBytes))
+	}
+	defer resp.Body.Close()
```
Comment (rnov) [inline resolver/contentResolver.go:113]: missing error handling

File: main_test.go [hunk 3]
```diff
@@ -29,6 +29,7 @@ const (
 	relationsResolverPath       = "/contentcollection/{uuid}/relations"
 	relationsResolverHealthPath = "/__health"
 	tid                         = "tid_test123456"
+	requestTimeoutInt           = 2000000000
```
Comment (rnov) [inline main_test.go]: suggestion: 2e9 might be more verbose ?

File: config.go [hunk 4]
```diff
@@ -19,6 +19,7 @@ type serviceConfig struct {
 	kafkaAddr                  *string
 	kafkaHostname              *string
 	kafkaAuth                  *string
+	requestTimeout             *int
```
Comment (rnov) [inline config.go:22]: what's the point in general to use pointers for unmutable values like string or ints ?

**Review by rnov — CHANGES_REQUESTED** [review]
as we agreed, all defer calls must handle errors a candidates to check  :
```
./forwarder/forwarder.go:48:	defer resp.Body.Close()
./forwarder/forwarder_test.go:35:		defer r.Body.Close()
./forwarder/forwarder_test.go:57:	defer mockServer.Close()
./forwarder/forwarder_test.go:68:	defer mockServer.Close()
./forwarder/forwarder_test.go:80:	defer mockServer.Close()
./relations/relationsResolver_test.go:24:	defer file.Close()
./relations/relationsResolver.go:33:	defer resp.Body.Close()
./healthcheck.go:129:	defer resp.Body.Close()
./common_test.go:50:	defer file.Close()
./unfolder.go:73:	defer req.Body.Close()
./unfolder_test.go:32:	defer server.Close()
./unfolder_test.go:54:	defer server.Close()
./unfolder_test.go:65:	defer resp.Body.Close()
./unfolder_test.go:86:	defer server.Close()
./unfolder_test.go:100:	defer resp.Body.Close()
./unfolder_test.go:127:	defer server.Close()
./unfolder_test.go:151:	defer resp.Body.Close()
./unfolder_test.go:176:	defer server.Close()
./unfolder_test.go:201:	defer resp.Body.Close()
./unfolder_test.go:230:	defer server.Close()
./unfolder_test.go:254:	defer resp.Body.Close()
./unfolder_test.go:285:	defer server.Close()
./unfolder_test.go:315:	defer resp.Body.Close()
./unfolder_test.go:345:	defer server.Close()
./unfolder_test.go:378:	defer resp.Body.Close()
./unfolder_test.go:404:	defer server.Close()
./unfolder_test.go:443:	defer resp.Body.Close()
./unfolder_test.go:461:	defer server.Close()
./unfolder_test.go:485:	defer resp.Body.Close()
./resolver/contentResolver.go:34:	defer resp.Body.Close()
./resolver/contentResolver.go:113:	defer resp.Body.Close()
./resolver/contentResolver_test.go:30:	defer file.Close()
./resolver/uuidResolver_test.go:86:	defer file.Close()
./main_test.go:39:	defer writerServer.Close()
./main_test.go:42:	defer contentResolverServer.Close()
./main_test.go:45:	defer relationsResolverServer.Close()
./main_test.go:52:	defer unfolderServer.Close()
./main_test.go:59:	defer resp.Body.Close()
./main_test.go:79:	defer writerServer.Close()
./main_test.go:82:	defer contentResolverServer.Close()
./main_test.go:85:	defer relationsResolverServer.Close()
./main_test.go:92:	defer unfolderServer.Close()
./main_test.go:99:	defer resp.Body.Close()
./main_test.go:119:	defer writerServer.Close()
./main_test.go:122:	defer contentResolverServer.Close()
./main_test.go:125:	defer relationsResolverServer.Close()
./main_test.go:132:	defer unfolderServer.Close()
./main_test.go:138:	defer resp.Body.Close()
./main_test.go:161:		defer r.Body.Close()
./main_test.go:67:	json.Unmarshal(body, &healthResult)
./main_test.go:107:	json.Unmarshal(body, &healthResult)

```

File: helm/content-collection-unfolder/app-configs/content-collection-unfolder_delivery.yaml (hunk 1 above)
Comment (gkazakov111) [reply helm/content-collection-unfolder/app-configs/content-collection-unfolder_delivery.yaml]: fixed

File: resolver/contentResolver.go (hunk 2 above)
Comment (gkazakov111) [reply resolver/contentResolver.go:113]: will be fixed generally later

File: main_test.go (hunk 3 above)
Comment (gkazakov111) [reply main_test.go]: fixed

**Review by rnov — DISMISSED** [review]

**Review by gotha — APPROVED** [review]

**Pull request merged**